index.bs

```diff
@@ -795,6 +797,11 @@ These events bubble up to SpeechSynthesis.
   No guarantee is given as to where charIndex will be with respect to word boundaries (such as at the end of the previous word or the beginning of the next word), only that all text before charIndex has already been spoken, and all text after charIndex has not yet been spoken.
   The user agent must return this value if the speech synthesis engine supports it, otherwise the user agent must return undefined.</dd>
 
+  <dt><dfn attribute for=SpeechSynthesisEvent>charLength</dfn> attribute</dt>
+  <dd>This attribute indicates the length of the text (word or sentence) that will be spoken corresponding to this event.
+  This attribute is the length, in characters, starting from this event's {{charIndex}}.
+  The user agent must return this value if the speech synthesis engine supports it or the user agent can otherwise determine it, otherwise the user agent must return undefined.</dd>
```

Comment:
The problem of returning undefined mentioned in https://github.com/w3c/speech-api/pull/30#discussion_r207848595 remains to be solved here. If you're also planning to implement this (in Chromium?) we can just do what the implementation will do. There are really only two options:
- Don't expose the `charLength` attribute if it's not supported, so that `'charLength' in SpeechSynthesisEvent.prototype` is false (makes feature detection possible)
- If this is a dynamic and depends on the speech synthesis engine at runtime, make the attribute nullable and say null here instead of undefined. There's some precedence for this in [WebRTC](https://w3c.github.io/webrtc-pc/#rtcerror-interface).

Reply:
I agree that nullable is the way to go, since it ends up being dynamic based on which speech synthesis engine you are using. Should this logic apply to charIndex, elapsedTime, and name as well? The current implementations seem to return the default values for these fields instead of undefined, and it might cause some compat issues if this is changed.

Reply:
Do you mean that it returns 0? If making the distinction between zero and unknown isn't important to clients of the API we could change the spec to do the same. Bull null is fine too.

Would you be up to sending a PR to change the other attributes to change in a similar way?